saleae/saleae.py

```diff
@@ -439,111 +447,132 @@ def load_from_file(self, file_path_on_target_machine):
+		'''CVS export analog/mixed: [HEADERS|NO_HEADERS], [COMMA|TAB], [BIN|DEC|HEX|ASCII], [VOLTAGE|ADC]'''
+
+		# Do argument verification
+		if delimeter.lower() not in ['comma', 'tab']: raise ValueError('bad CSV delimeter')
+		if display_base.lower() not in ['bin', 'dec', 'hex', 'ascii']: raise ValueError('bad CSV display base')
+		if analog_format.lower() not in ['voltage', 'adc']: raise ValueError('bad CSV analog format')
+
+		# Build arguments
+		self._build('HEADERS' if column_headers else 'NO_HEADERS')
+		self._build(delimeter.upper())
+		self._build(display_base.upper())
+		self._build(analog_format.upper())
+
+	def _export_data_digital_csv(self, column_headers=True, delimeter='comma', timestamp='time_stamp', display_base='hex', rows_per_change=True):
+		'''CVS export digital: [HEADERS|NO_HEADERS], [COMMA|TAB], [TIME_STAMP|SAMPLE_NUMBER], [COMBINED, [BIN|DEC|HEX|ASCII]|SEPARATE], [ROW_PER_CHANGE|ROW_PER_SAMPLE]'''
+
+		# Do argument verification
+		if delimeter.lower() not in ['comma', 'tab']: raise ValueError('bad CSV delimeter')
+		if timestamp.lower() not in ['time_stamp', 'sample_number']: raise ValueError('bad CSV timestamp')
+		if display_base.lower() not in ['bin', 'dec', 'hex', 'ascii', 'separate']: raise ValueError('bad CSV display base')
+
+		# Build arguments
+		self._build('HEADERS' if column_headers else 'NO_HEADERS')
+		self._build(delimeter.upper())
+		self._build(timestamp.upper())
+		self._build('SEPERATE' if display_base.lower() == 'SEPERATE' else ['COMBINED', display_base.upper()])
+		self._build('ROW_PER_CHANGE' if rows_per_change else 'ROW_PER_SAMPLE')
+
+	def _export_data_digital_vcd(self):
+		'''VCD digital: no arguments'''
+		pass
+
+	def _export_data_analog_matlab(self, analog_format='voltage'):
+		'''Matlab analog: [VOLTAGE|ADC]'''
+
+		# Do argument verification
+		if analog_format.lower() not in ['voltage', 'adc']: raise ValueError('bad Matlab analog format')
+
+		# Build arguments
+		self._build(analog_format.upper())
 
+	def _export_data_digital_matlab(self):
+		'''Matlab digital: no arguments'''
+		pass
+
+
+	def export_data(self, file_path_on_target_machine, digital_channels=None, analog_channels=None, time_span=None, format='csv', **export_args):
+		'''Export command:
+			EXPORT_DATA2,
+			<filename>,
+			[ALL_CHANNELS|SPECIFIC_CHANNELS, [DIGITAL_ONLY|ANALOG_ONLY|ANALOG_AND_DIGITAL], <channel index> [ANALOG|DIGITAL], ..., <channel index> [ANALOG|DIGITAL]],
+			[ALL_TIME|TIME_SPAN, <(double)start>, <(double)end>],
+			[BINARY, <binary settings>|CSV, <csv settings>|VCD|MATLAB, <matlab settings>]
+		'''
 		while not self.is_processing_complete():
 			time.sleep(1)
 
-		# The path needs to be absolute. This is hard to check reliably since we
-		# don't know the OS on the target machine, but we can do a basic check
-		# for something that will definitely fail
+		# Saleae Logic should resolve relative paths, I do not see reasons not to do this ...
```

Comment:
Saleae Logic software does not (currently) resolve relative paths. The absolute requirement is from saleae.